api/contrib/envoy/extensions/filters/network/postgres_proxy/v3alpha/postgres_proxy.proto

```diff
@@ -20,6 +20,21 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#extension: envoy.filters.network.postgres_proxy]
 
 message PostgresProxy {
+  // Upstream SSL operational modes.
+  enum SSLMode {
+    // Do not encrypt upstream connection to the server.
+    SSL_DISABLE = 0;
+
+    // Establish upstream SSL connection to the server. If the server does not
+    // accept the request for SSL connection, the session is terminated.
+    SSL_REQUIRE = 1;
+
+    // Try to establish upstream SSL connection to the server. If the server does not
+    // accept the request for SSL or Envoy cannot successfully establish secure connection
+    // to upstream, the session continues in clear-text.
+    SSL_PREFER = 2;
```

Comment:
What's the use case or motivation of this one? From a security perspective this is no better than disabled SSL, and subject to MITM to disable SSL completely.

Reply:
It came from #19527. But I agree with you that it would be easy to mount MITM. 
So the remaining options are only REQUIRE and DISABLE. Enum is not truly necessary and boolean can be used. I will refactor the code. Thanks!

Reply:
We can leave it as an enum to be future proof.

Reply:
Done. Enums are used.